gapii/cc/call_observer.cpp

```diff
@@ -118,18 +118,31 @@ void CallObserver::observePending() {
 }
 
 void CallObserver::enter(const ::google::protobuf::Message* cmd) {
+    if (!mSpy->should_trace(mApi)) {
+        return;
+    }
     mEncoderStack.push(encoder()->group(cmd));
 }
 
 void CallObserver::encode(const ::google::protobuf::Message* cmd) {
+    if (!mSpy->should_trace(mApi)) {
+        return;
+    }
     encoder()->object(cmd);
 }
 
 void CallObserver::exit() {
+    if (!mSpy->should_trace(mApi)) {
```

Comment:
This scares me. If the `should_trace` return value changes between `enter` and `exit` then You're Going To Have A Bad Time. 
Are there cases where `should_trace` should change for the lifetime of a `CallObserver`? If not, can we fetch it once and store it as a member?

Reply:
There are no cases (that I know) that would make this change during a call.
The only place we change ShouldTrace is during MEC start

Reply:
I can definitely store it off as a member, that will prevent this from accidentally happening.